utilities/javascript/mergeSort.js

```diff
@@ -0,0 +1,54 @@
+function mergeSort(arr) {
+  if (arr.length < 2) {
+    // Arrays of length 0 or 1 are sorted by definition
+    return arr;
+  }
+
+  const left = arr.slice(0, Math.floor(arr.length / 2));
+  const right = arr.slice(Math.floor(arr.length / 2), Math.floor(arr.length));
+
+  return merge(mergeSort(left), mergeSort(right));
+}
+
+function merge(arr1, arr2) {
+  const merged = [];
+
+  let i = 0;
+  let j = 0;
+
+  while (i < arr1.length && j < arr2.length) {
+    if (arr1[i] < arr2[j]) {
+      merged.push(arr1[i]);
+      i++;
+    } else if (arr2[j] < arr1[i]) {
+      merged.push(arr2[j]);
+      j++;
+    } else {
+      merged.push(arr1[i]);
+      i++;
+      j++;
+    }
+  }
+
+  while (i < arr1.length) {
```

Comment:
Instead of:
```
 +  while (i < arr1.length) {
 +    merged.push(arr1[i]);
 +    i++;
 +  }
 +  while (j < arr2.length) {
 +    merged.push(arr2[j]);
 +    j++;
 +  }

```
This is shorter:

```
merged.push(...arr1.slice(i), ...arr2.slice(j));
```